card/lib/card/set/event.rb

```diff
@@ -0,0 +1,127 @@
+        when opts[:after] || opts[:before]
+          # ignore :in options
+        when opts[:in]
+          opts[:after] = :"#{opts.delete(:in)}_stage" if opts[:in]
+        end
+        opts[:on] = [:create, :update] if opts[:on] == :save
+      end
+
+      def define_event_delaying_method event, method_name
+        class_eval do
+          define_method method_name, proc {
+            s_attr =
+              serializable_attributes.each_with_object({}) do |name, hash|
+                value = instance_variable_get("@#{name}")
+                hash[name] =
+                  # ActiveJob doesn't accept symbols as arguments
+                  if Symbol === value
+                    { value: value.to_s, symbol: true }
+                  else
+                    { value: value }
+                  end
+              end
+            Object.const_get(event.to_s.camelize).perform_later(self, s_attr)
+          }
+        end
+      end
+
+      def define_event_method event, call_method, _opts
+        class_eval do
+          define_method event do
+            # puts "#{event}".green
+            run_callbacks event do
+              send call_method
+            end
+          end
+        end
+      end
+
+      # creates an Active Job.
+      # The scheduled job gets the card object as argument and all serializable
+      # attributes of the card.
+      # (when the job is executed ActiveJob fetches the card from the database so
+      #  all attributes get lost)
+      # @param name [String] the name for the ActiveJob child class
+      # @param final_method [String] the name of the card instance method to be
+      #   queued
+      # @option queue [Symbol] (:default) the name of the queue8
+      def define_active_job name, final_method, queue=:default
+        class_name = name.to_s.camelize
+        eval %{
+        class ::#{class_name} < ActiveJob::Base
+          queue_as #{queue}
+        end
+      }
+        Object.const_get(class_name).class_eval do
+          define_method :perform, proc { |card, attributes|
+            attributes.each do |attname, args|
+              # symbols are not allowed so all symbols arrive here as strings
+              # convert strings that were symbols before back to symbols
+              value = args[:symbol] ? args[:value].to_sym : args[:value]
+              card.instance_variable_set("@#{attname}", value)
+            end
+            card.include_set_modules
+            card.send final_method
+          }
+        end
+      end
+
+      def set_event_callbacks event, opts
+        [:before, :after, :around].each do |kind|
+          if (object_method = opts.delete(kind))
+            this_set_module = self
+            Card.class_eval do
+              set_callback(
+                object_method, kind, event,
+                prepend: true, if: proc do |c|
+                c.singleton_class.include?(this_set_module) &&
+                  c.event_applies?(opts)
+              end
```

Comment:
`end` at 119, 14 is not aligned with `proc do |c|` at 116, 35 or `prepend: true, if: proc do |c|` at 116, 16.